static_frame/core/index_hierarchy.py

```diff
@@ -1051,14 +1050,65 @@ def gen() -> tp.Iterator[tp.Tuple[tp.Hashable, ...]]:
                 )
 
 
-    # def relabel_at_depth(self: IH,
-    #         mapper: RelabelInput,
-    #         depth_level: DepthLevelSpecifier = 0
-    #         ) -> IH:
-    #     '''
-    #     Return a new :obj:`IndexHierarchy` after applying the hte `mapper` to the depth level or levels specified by `depth_level`.
-    #     '''
-    #     pass
+    def relabel_at_depth(self,
+            mapper: RelabelInput,
+            depth_level: DepthLevelSpecifier = 0
+            ) -> "IndexHierarchy":
+        '''
+        Return a new :obj:`IndexHierarchy` after applying the the `mapper` to the depth level or levels specified by `depth_level`.
+        '''
+        if self._recache:
+            self._update_array_cache()
+
+        if isinstance(depth_level, INT_TYPES):
+            # Single depth levels have an optimized path through IndexLevels
+            return self.__class__(
+                    levels=self._levels.relabel_at_specific_depth(mapper, depth_level),
+                    name=self._name,
+                    )
+
+        depth_level = list(depth_level)
+
+        if len(depth_level) == 1:
+            # Optimized path for single depth level
+            return self.__class__(
+                levels=self._levels.relabel_at_specific_depth(mapper, depth_level[0]),
+                name=self._name,
+                )
+        elif len(depth_level) == self.depth:
+            # Equivalent to relabel!
+            return self.relabel(mapper)
```

Comment:
What if the `depth_level` is invalid (i.e., `[1, 1, 1]`) or in a different order than expected?

Reply:
Ah I didn't consider that edge case. Good suggestion